tests/unit/test_model_setup.py

```diff
@@ -0,0 +1,904 @@
+"""
+This file contains tests that were constructed to check the correct assingment of parameters
+in standard models in PorePy.
+
+For each model, two main methods are constructed: one that checks default parameters and
+another that checks non-default parameters. For the latter, three different sets of
+non-default parameters for a mixed-dimensional Cartesian grid containing a single vertical
+fracture have been written.
+
+Moreover, for each model, the parameters are checked at two stages, namely before running
+the model and after running the model.
+
+Note that only parameters belonging to the specific class are tested. This means that
+inherited parameters are not tested as this would be too cumbersome (and redundant).
+However, this naturally requires writing tests for all present models.
+
+Todo:
+    Create tests for the rest of the models.
+"""
+
+import porepy as pp
+import pytest
+import numpy as np
+
+from typing import Union
+
+
+class TestContactMechanicsBiot:
```

Comment:
I really appreciate the effort to test the models, but I don't think the approach taken here is a good one: The code is so long that it begs for a better solution, in particular considering that this is actually not our complex model.

I have some (vague) ideas on how this can be realized in a better manner; I suggest we discuss this @jhabriel

Reply:
Point taken. I also underestimated how long turned out to be to test the _non-default_ parameters of a model. It becomes ugly very fast.

Reply:
Should we remove the new test and try to merge this branch?

Reply:
I'm OK with this. Let me create an issue so we don't forget to incorporate the test in the (near) future.

Reply:
Are you OK with merging this PR without the tests for the models @keileg? I've created a separate issue for this, see #685.